modules/router-certificates.adoc

```diff
@@ -0,0 +1,18 @@
+// Module included in the following assemblies:
+//
+// * authentication/understanding-certificates.adoc
+
+[id="router-certificates_{context}"]
+= Router certificates
+
```

Comment:
@Miciah As discussed earlier, I stubbed out this section where we can build out the information that the customer is requesting. Thanks!

Reply:
@Miciah Can you please provide some details by end of week on router certs? We are needing:
* The purpose
* File path
* Default expiration term
* How to set custom expiration term
* How to specify the expiration date of all certificates used in OpenShift when installing OpenShift
* Which services use it
* How to update/extend it
* Will certificates that are about to expire be automatically renewed by the operator?

Reply:
@ahardin-rh, how's this?

> * The purpose

The Ingress Operator uses certificates for the following purposes:

* Securing access to metrics for Prometheus.
* Securing access to Routes.

> * File path

To secure access to Ingress Operator and Ingress Controller metrics, the Ingress Operator uses [service serving certificates](../../blob/master/authentication/certificates/service-serving-certificate.adoc#understanding-service-serving_{context}). The Operator requests a certificate from the `service-ca` controller for its own metrics, and the `service-ca` controller puts the certificate in a secret named `metrics-tls` in the `openshift-ingress-operator` namespace.  Additionally, the Ingress Operator requests a certificate for each Ingress Controller, and the `service-ca` controller puts the certificate in a secret named `router-metrics-certs-<name>` (where `<name>` is the name of the Ingress Controller) in the `openshift-ingress` namespace.

Each Ingress Controller has a default certificate that the Controller uses for [secured routes](../../blob/master/networking/routes/secured-routes.adoc) that do not specify their own certificates.  You can [set a custom default certificate](../../blob/master/networking/ingress-operator.adoc#nw-ingress-setting-a-custom-default-certificate_{context}), or else the Operator generates a certificate for you.  The Operator uses its own self-signed signing certificate to sign any default certificate that it generates.  The Operator generates this signing certificate and puts it in a secret named `router-ca` in the `openshift-ingress-operator` namespace.  When the Operator generates a default certificate, it puts the default certificate in a secret named `router-certs-<name>` (where `<name>` is the name of the Ingress Controller) in the `openshift-ingress` namespace.

:warning: The Operator generates a default certificate for an Ingress Controller to serve as a placeholder till you configure a custom default certificate.  Do not use operator-generated default certificates in production clusters.

> * Default expiration term

The expiration terms for the Ingress Operator's certificates are as follow:

* The expiration date for metrics certificates that the `service-ca` controller creates is 2 years after the date of creation.
* The expiration date for the Operator's signing certificate is 2 years after the date of creation.
* The expiration date for default certificates that the Operator generates is 2 years after the date of creation.

> * How to set custom expiration term

You cannot specify custom expiration terms on certificates that the Ingress Operator or `service-ca` controller creates.

> * How to specify the expiration date of all certificates used in OpenShift when installing OpenShift

You cannot specify expiration terms when installing OpenShift for certificates that the Ingress Operator or `service-ca` controller create.

> * Which services use it

Prometheus uses the certificates that secure metrics.

The Ingress Operator uses its signing certificate to sign default certificates that it generates for Ingress Controllers for which you do not set [custom default certificates](../../blob/master/networking/ingress-operator.adoc#nw-ingress-setting-a-custom-default-certificate_{context}).

Cluster components that use [secured routes](../../blob/master/networking/routes/secured-routes.adoc) may use the default Ingress Controller's default certificate.

Ingress to the cluster via a secured route uses the default certificate of the Ingress Controller via which the route is accessed unless the route specifies its own certificate.

> * How to update/extend it

To force regeneration of a certificate, delete the corresponding secret.

> * Will certificates that are about to expire be automatically renewed by the operator?

The `service-ca` controller automatically rotates the certificates that it issues.  However, it is possible to [manually rotate service serving certificates](../../blob/master/authentication/certificates/service-serving-certificate.adoc#manually-rotate-service-ca_{context}).

The Ingress Operator does not rotate its own signing certificate or the default certificates that it generates.  Operator-generated default certificates are intended as placeholders for [custom default certificates](../../blob/master/networking/ingress-operator.adoc#nw-ingress-setting-a-custom-default-certificate_{context}) that you configure.